src/components/common/Content/MovieTVShowPageContent/MovieTVShowPageContent.tsx

```diff
@@ -0,0 +1,61 @@
+import './MovieTVShowPageContent.css'
+import { useLocation } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { MOVIE_PAGE_CONTENT_KEY } from "../../../../data/local-storage-keys.ts";
+import { MovieTvPageContent } from "../../../../interfaces /MoviePageResponseInterface.ts";
+
+export const MovieTVShowPageContent = () => {
+    const location = useLocation()
+    const { id, type } = location.state
+    const [movieTvPageData, setMovieTvPageData] = useState((): MovieTvPageContent => {
+        const storedData = localStorage.getItem(MOVIE_PAGE_CONTENT_KEY);
+        return storedData ? JSON.parse(storedData)[0] : null;
+    })
+
+    useEffect(() => {
+        const itemId = id ? parseInt(id) : null
+
+        if (movieTvPageData && itemId === movieTvPageData.id) {
+            const storedData = localStorage.getItem(MOVIE_PAGE_CONTENT_KEY);
+
+            if (storedData && JSON.parse(storedData).length === 0) {
+                setMovieTvPageData(JSON.parse(storedData)[0]);
+            }
+        }
+    }, [id, movieTvPageData, type])
+
+    return (
+        <div className="movie-tv-page space-between-row-center-flex">
```

Comment:
Also you need to understand architecture patterns in front-end development. Here you're mapping data right in component. It's bad practice. Start you diving in architecture patterns with this article https://habr.com/ru/articles/726354/